### PR TITLE
feat: use latest llvm for stage1 in musl builds

### PR DESCRIPTION
**Pull request by antonbaliasnikov**
# What ❔

Use the latest release of LLVM for stage 1 in Musl builds.

## Why ❔

To solve build issues caused by our fork breaking other LLVM targets.

## Checklist

- [x] PR title corresponds to the body of PR.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `cargo fmt` and checked with `cargo clippy`.

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -12,6 +12,12 @@ const DRY_RUN: bool = false;
 /// Enable verbose output.
 pub const VERBOSE: bool = false;
 
+/// The LLVM host repository URL.
+pub const LLVM_HOST_SOURCE_URL: &str = "https://github.com/llvm/llvm-project";
+
+/// The LLVM host repository tag.
+pub const LLVM_HOST_SOURCE_TAG: &str = "llvmorg-18.1.4";
```
Comment (antonbaliasnikov) [inline src/utils.rs:19]: I put the latest release tag here for now. Please, let me know if we would like to pin some other version.

File: src/platforms/x86_64_linux_musl.rs [hunk 2]
```diff
@@ -213,15 +214,7 @@ fn build_crt(
                 "-DCMAKE_C_COMPILER='clang'",
                 "-DCMAKE_CXX_COMPILER='clang++'",
                 "-DLLVM_ENABLE_PROJECTS='compiler-rt'",
-                format!(
```
Comment (antonbaliasnikov) [inline src/platforms/x86_64_linux_musl.rs:216]: It's not required for compiler-rt, we need just host targets here.

File: src/lib.rs [hunk 3]
```diff
@@ -17,12 +17,46 @@ use std::collections::HashSet;
 use std::path::PathBuf;
 use std::process::Command;
 
+///
+/// Executes the LLVM host repository cloning for stage 1 MUSL builds.
+///
+pub fn clone_host() -> anyhow::Result<()> {
+    let destination_path = PathBuf::from(LLVMPath::DIRECTORY_LLVM_HOST_SOURCE);
+    if destination_path.exists() {
+        println!(
```
Comment (antonbaliasnikov) [inline src/lib.rs:26]: As we don't use llvm.lock for the host, I believe it's good enough to just print a warning and skip if it's already cloned. But let me know if you have some other thoughts.

**Comment by antonbaliasnikov** [comment]
@hedgar2017 , I did not increment a version, because we already incremented it in the previous PR, I guess it should be ok to combine changes.

File: tests/common/mod.rs [hunk 4]
```diff
@@ -4,7 +4,7 @@ pub const ZKEVM_LLVM: &str = "zkevm-llvm";
 pub const PACKAGE_VERSION: &str = env!("CARGO_PKG_VERSION");
 pub const ERA_LLVM_REPO_URL: &str = "https://github.com/matter-labs/era-compiler-llvm";
 pub const ERA_LLVM_REPO_TEST_BRANCH: &str = "v1.4.2";
-pub const ERA_LLVM_REPO_TEST_REF: &str = "714cc6aa05ea03dab3e4e504cb8c25973a33ea94";
+pub const ERA_LLVM_REPO_TEST_REF: &str = "714cc6aa0";
```
Comment (hedgar2017) [inline tests/common/mod.rs:7]: Can we use the full commit hash here?
It's easy to get it via `git log` and the length of these short refs is really unpredictable...

File: src/platforms/x86_64_linux_musl.rs [hunk 5]
```diff
@@ -213,15 +214,7 @@ fn build_crt(
...
-                    "-DLLVM_TARGETS_TO_BUILD='{}'",
-                    targets
-                        .into_iter()
-                        .map(|platform| platform.to_string())
-                        .collect::<Vec<String>>()
-                        .join(";")
-                )
-                .as_str(),
+                "-DLLVM_TARGETS_TO_BUILD='X86'",
```
Comment (hedgar2017) [inline src/platforms/x86_64_linux_musl.rs:217]: We have a constant with `to_string()` defined in the `Platform` enum.

File: src/utils.rs (hunk 1 above)
Comment (hedgar2017) [reply src/utils.rs:19]: Let's grab the same 17.0.6 just to reduce variability?

File: src/platforms/aarch64_linux_musl.rs [hunk 6]
```diff
@@ -213,15 +214,7 @@ fn build_crt(
...
-                format!(
-                    "-DLLVM_TARGETS_TO_BUILD='{}'",
-                    targets
-                        .into_iter()
-                        .map(|platform| platform.to_string())
-                        .collect::<Vec<String>>()
-                        .join(";")
-                )
-                .as_str(),
+                "-DLLVM_TARGETS_TO_BUILD='AArch64'",
```
Comment (hedgar2017) [inline src/platforms/aarch64_linux_musl.rs:217]: Also const from `Platform` please.

File: src/lib.rs (hunk 3 above)
Comment (hedgar2017) [reply src/lib.rs:26]: Use `eprintln!` please so it writes to `stderr`.

**Review by hedgar2017 — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 3 above)
Comment (antonbaliasnikov) [reply src/lib.rs]: Done.

File: tests/common/mod.rs (hunk 4 above)
Comment (antonbaliasnikov) [reply tests/common/mod.rs]: Yeah, fixed the tests code.

File: src/platforms/x86_64_linux_musl.rs (hunk 5 above)
Comment (antonbaliasnikov) [reply src/platforms/x86_64_linux_musl.rs]: Awesome, updated these lines.

File: src/platforms/aarch64_linux_musl.rs (hunk 6 above)
Comment (antonbaliasnikov) [reply src/platforms/aarch64_linux_musl.rs]: Done.

**Review by hedgar2017 — APPROVED** [review]

**Pull request merged**